### PR TITLE
fix(gomod): skip updating import path for incompatible versions

### PR DESCRIPTION
**Pull request by maxbrunet**
## Changes

Skip `gomodUpdateImportPaths` post update option if `newVersion` ends with `+incompatible`

## Context

> `+incompatible` indicates that a version is part of the same module as versions with lower major version numbers;

https://go.dev/ref/mod#incompatible-versions

Renovate opened this PR: parca-dev/parca-agent#1427

### Got

```diff
-	github.com/docker/docker v20.10.23+incompatible
+	github.com/docker/docker/v23 v23.0.1+incompatible
```

### Expected

```diff
-	github.com/docker/docker v20.10.23+incompatible
+	github.com/docker/docker v23.0.1+incompatible
```

## Documentation (please check one with an [x])

- [ ] I have updated the documentation, or
- [x] No documentation update is required

## How I've tested my work (please select one)

I have verified these changes via:

- [ ] Code inspection only, or
- [ ] Newly added/modified unit tests, or
- [ ] No unit tests but ran on a real repository, or
- [x] Both unit tests + ran on a real repository

### CODE REVIEW

**Comment by rarkins** [comment]
Have you verified this on a real repo?

File: lib/modules/manager/gomod/artifacts.spec.ts [hunk 1]
```diff
@@ -1434,6 +1434,52 @@ describe('modules/manager/gomod/artifacts', () => {
     ]);
   });
 
+  it('skips updating import paths when incompatible version', async () => {
+    fs.readLocalFile.mockResolvedValueOnce('Current go.sum');
```
Comment (viceice) [inline lib/modules/manager/gomod/artifacts.spec.ts:1438]: please chain like below

File: lib/modules/manager/gomod/artifacts.ts [hunk 2]
```diff
@@ -129,14 +129,19 @@ function getUpdateImportPathCmds(
   }
 
   const updateImportCommands = updatedDeps
-    .filter(({ newVersion }) => valid(newVersion))
+    .filter(
+      ({ newVersion }) =>
+        valid(newVersion) &&
+        newVersion !== undefined &&
```
Comment (viceice) [inline lib/modules/manager/gomod/artifacts.ts:135]: isn't `valid()` already checking that?

**Comment by maxbrunet** [comment]
> Have you verified this on a real repo?

Good call, no I had not, and did catch something, working now: https://github.com/maxbrunet/renovate-20812/pull/1

File: lib/modules/manager/gomod/artifacts.spec.ts [hunk 3]
```diff
@@ -1434,6 +1434,53 @@ describe('modules/manager/gomod/artifacts', () => {
...
   });
 
+  it('skips updating import paths when incompatible version', async () => {
+    fs.readLocalFile
+      .mockResolvedValueOnce('Current go.sum')
+      .mockResolvedValueOnce(null); // vendor modules filename
+    const execSnapshots = mockExecAll();
+    git.getRepoStatus.mockResolvedValueOnce({
+      modified: ['go.sum'],
+    } as StatusResult);
```
Comment (viceice) [inline lib/modules/manager/gomod/artifacts.spec.ts:1444]: use our partial helper, see test below

File: lib/modules/manager/gomod/artifacts.ts (hunk 2 above)
Comment (viceice) [reply lib/modules/manager/gomod/artifacts.ts:135]: yes

**Review by viceice — APPROVED** [review]

**Pull request merged**